### DISCUSSION TITLE
Which Java version should we target?

### DISCUSSION THREAD

**Discussion by madsop-nav** [openhtmltopdf/openhtmltopdf#45] [Ideas]
As of now, we target Java 8. As it's now more than 10 years and 13 versions since 8 was released, I suggest that we start targeting a newer Java version.

What are your thoughts?

**Comment by burka** [comment]
We are using openhtmltopdf with Java 17 and Java 21. 17 or anything below would work for us.

It should be considered that this is an old lib with potentially a lot of longtime users with older projects and there is still [Extended Support available](https://www.oracle.com/de/java/technologies/java-se-support-roadmap.html) for Java 8, 11, 17, 21.   
But - whoever doesn't update Java probably doesn't update Openhtmltopdf, so if there is some actually used benefit in that MR, increase the requrirement :+1:

**Reply by kjetilod** [reply]
I think supporting the current and previous LTS is enough, so requiring Java 17 should be fine 👍

**Comment by lagar84** [comment]
Unless needed, stick with the java version pdfbox needs? (java 8 so far). As a library, the least you require from users, the better.
